source/src/protocols/cyclic_peptide/crosslinker/lanthionine_util.cc

```diff
@@ -0,0 +1,620 @@
+		|| pose.residue_type(dalares).base_name() == "DALA" || pose.residue_type(dalares).base_name() == "ALA",
+		errmsg + "Position " + std::to_string( dalares ) + " was selected as the non-cys residue for a "
+		"lanthionine connection, but this residue is not a D-ALA or ALA or Butyrine Variant."
+	);
+
+	//if ( pose.residue_type(dalares).is_lower_terminus() ) {
+	// core::pose::remove_lower_terminus_type_from_pose_residue( pose, dalares );
+	// TR << "Removed lower terminus type from residue " << pose.residue_type(dalares).base_name() << dalares << "." << std::endl;
+	//}
+
+	core::pose::add_variant_type_to_pose_residue( pose, core::chemical::SIDECHAIN_CONJUGATION, cysres );
+	TR << "Added sidechain conjugation type to residue " << pose.residue_type(cysres).base_name() << cysres << "." << std::endl;
+
+	core::pose::add_variant_type_to_pose_residue( pose, core::chemical::SC_BRANCH_POINT, dalares );
+	TR << "Added acetylated SC_BRANCH_POINT connection type to residue " << pose.residue_type(dalares).base_name() << dalares << "." << std::endl;
+
+	pose.update_residue_neighbors();
+	// This connects to ii+1 by default, but that's appropriate -- it just got this connection harmed
+	// by adding the patch anyway.
+	//pose.conformation().update_polymeric_connection( dalares );
+
+	// Force rebuild of the "H":
+	//if ( pose.residue_type(dalares).has("H") ) {
+	// pose.set_xyz( core::id::NamedAtomID( "H", dalares ), pose.residue_type( dalares ).icoor( pose.residue_type( dalares ).atom_index("H") ).build( pose.residue( dalares ), pose.conformation() ) );
+	//}
+
+	//pose.update_residue_neighbors();
+}
+
+/// @brief Set up the mover that creates lanthionine lariat bonds.
+void
+set_up_lanthionine_bond_mover(
+	protocols::simple_moves::DeclareBond & termini,
+	core::pose::Pose const & pose,
+	core::Size const dalares,
+	core::Size const cysres
+) {
+	//Get the name of the first sidechain connection atom:
+	core::chemical::ResidueType const & restype( pose.residue_type( dalares ) );
+	core::Size const dalares_sc_connection_id( restype.n_possible_residue_connections() ); //We will assume that the highest-numbered residue connection is the sidechain connection ID.
+	core::Size const dalares_sc_connection_atom_index( restype.residue_connect_atom_index( dalares_sc_connection_id ) );
+	std::string const dalares_sc_connection_atom( restype.atom_name( dalares_sc_connection_atom_index ) );
+
+	//Get the name of the second sidechain connection atom:
+	core::chemical::ResidueType const & restype2( pose.residue_type( cysres ) );
+	core::Size const cysres_sc_connection_id( restype2.n_possible_residue_connections() ); //We will assume that the highest-numbered residue connection is the sidechain connection ID.
+	core::Size const cysres_sc_connection_atom_index( restype2.residue_connect_atom_index( cysres_sc_connection_id ) );
+	std::string const cysres_sc_connection_atom( restype2.atom_name( cysres_sc_connection_atom_index ) );
```

Comment:
Would it be worth checking this with the disulfide_atom_name() results?

Reply:
I am unsure. This code is all modeled off the thioether_util.cc code.